cdci_data_analysis/analysis/tokenHelper.py

```diff
@@ -59,10 +61,13 @@ def get_token_user_fail_email(decoded_token):
     return decoded_token.get('msfail', True) # TODO: make server configurable
 
 
-def get_decoded_token(token, secret_key):
+def get_decoded_token(token, secret_key, validate_token=True):
     # decode the encoded token
     if token is not None:
-        return jwt.decode(token, secret_key, algorithms=[default_algorithm])
+        if validate_token:
+            return jwt.decode(token, secret_key, algorithms=[default_algorithm])
+        else:
+            return json.loads(base64.b64decode(token.split(".")[1]+"=").decode())
```

Comment:
I am not sure this is guaranteed to always work. It was a simple trick useful in client when jwt is not installed. But in dispatcher, one should use proper jwt decode without validation.